src/voice/connection.rs

```diff
@@ -676,8 +678,12 @@ fn encryption_key(client: &mut WsClient) -> Result<Key> {
                     return Err(Error::Voice(VoiceError::VoiceModeInvalid));
                 }
 
-                return Key::from_slice(&desc.secret_key)
-                    .ok_or(Error::Voice(VoiceError::KeyGen));
+                if desc.secret_key.len() == KEY_SIZE {
+                    let key = Key::from_slice(&desc.secret_key);
+                    return Ok(XSalsa20Poly1305::new(key));
+                } else {
+                    return Err(Error::Voice(VoiceError::KeyGen));
+                }
```

Comment:
This pattern is unfortunate. Many of our other RustCrypto crates have a `new_var` method for this purpose. I'll look into getting that added to the upstream trait.

Reply:
Added a slice-based cipher init API here, although it might be awhile before we ship it:

https://github.com/RustCrypto/traits/pull/191

If you'd like I can add a TODO

Reply:
This seems fine for now.

Reply:
Sorry, this went through while I was reviewing. A TODO/FIXME would be welcome then, thanks.

Reply:
Added